src/automaton/core/network/rpc_server.cc

```diff
@@ -0,0 +1,107 @@
+#include "automaton/core/network/rpc_server.h"
+#include <cstdlib>
+#include <iostream>
+#include <string>
+#include <thread>
+#include <boost/asio/basic_stream_socket.hpp>
+#include <boost/asio/buffer.hpp>
+#include <boost/asio/error.hpp>
+#include <boost/asio/io_service.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/placeholders.hpp>
+#include <boost/asio/write.hpp>
+#include <boost/bind.hpp>
+#include "automaton/core/io/io.h"
+
+using boost::asio::ip::tcp;
+
+
+namespace automaton {
+namespace core {
+namespace network {
+
+
+session::session(boost::asio::io_service& io_service) // NOLINT
+  : socket_(io_service) {
+}
+
+tcp::socket& session::socket() {
+  return socket_;
+}
+
+void session::start() {
+  socket_.async_read_some(boost::asio::buffer(data_, kBufferSize),
+    boost::bind(&session::handle_read, this,
+      boost::asio::placeholders::error,
+      boost::asio::placeholders::bytes_transferred));
+}
+
+void session::handle_read(const boost::system::error_code& error, size_t bytes_transferred) {
+  if (!error) {
+    std::cout << std::string(data_, bytes_transferred) << std::endl;
+    boost::asio::async_write(socket_,
+      boost::asio::buffer(data_, bytes_transferred),
```

Comment:
instead of just sending the input back - make it so it calls the handler so it's handler(data)